src/libcore/cell.rs

```diff
@@ -280,6 +281,19 @@ unsafe impl<T> Send for Cell<T> where T: Send {}
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<T> !Sync for Cell<T> {}
 
+#[unstable(feature = "move_cell", issue = "39264")]
+impl<T:Clone> Clone for Cell<T> {
+    #[inline]
+    default fn clone(&self) -> Cell<T> {
+        let temp = unsafe { mem::uninitialized() };
+        let inner = self.replace(temp);
+        // If `clone` panics, let it crash.
```

Comment:
@F001: You'll probably want to use something similar to the `Hole` structure [described in the nomicon](https://github.com/rust-lang/rust/blob/master/src/doc/nomicon/exception-safety.md), which writes a valid value back into the original cell in case of a panic.